scripts/code_size_compare.py

```diff
@@ -51,15 +102,61 @@ def __init__(self, old_revision, new_revision, result_dir):
 
         self.old_rev = old_revision
         self.new_rev = new_revision
+        self.arch = arch
+        self.config = config
         self.git_command = "git"
-        self.make_command = "make"
+        self.pre_build_commands = PRE_BUILD_CMDS[config]
+        self.make_command = ''
+        self._set_make_command()
+
+        self.old_sizes = {}
+        self.new_sizes = {}
+        self.change_pcts = {}
 
     @staticmethod
     def validate_revision(revision):
         result = subprocess.check_output(["git", "rev-parse", "--verify",
                                           revision + "^{commit}"], shell=False)
         return result
 
+    def _set_make_command(self):
+        """Use the config and arch options passed to the object to determine
+           and set the correct build command."""
+        if self.arch == 'x86' and (self.config in {'default', 'full',\
+                                                   'baremetal'}):
+            self.make_command = 'make -j lib'
+            return
+
+        # Default just takes the current config, which may or may not work
+        # with baremetal targets. Warn the user.
+        if self.config == 'default':
+            print("Assuming that the current config is compatible with \
+                   baremetal targets. If it isn't the build may fail!")
+
+        if self.config in {'default', 'baremetal'}:
+            if self.arch == 'aarch32':
+                self.make_command = 'make lib CC=armclang CFLAGS=\
+                                    \"--target=arm-arm-none-eabi \
+                                    -mcpu=cortex-m33 -Os'
+            if self.arch == 'aarch64':
+                self.make_command = 'make lib CC=armclang CFLAGS=\
```

Comment:
I got an error with this option. 
```
$ ./scripts/code_size_compare.py -o code-size-tracking -a aarch64 -c baremetal
Creating git worktree for 67b12585c19aaf437f885135bd45df68a161e373
The following command: make lib CC=armclang CFLAGS=                                    "--target=aarch64-arm-none-eabi failed and exited with code 2
Process output:
 b'/bin/sh: 1: Syntax error: Unterminated quoted string\n'
Removing temporary worktree ./temp-67b12585c19aaf437f885135bd45df68a161e373
```

It looks we missed latter `\"` for CFLAGS.

Reply:
@yanrayw do you also get the error for the `aarch32` option above it?

Reply:
@aditya-deshpande-arm There seems to be a few of these still causing issues. 
For `-c` options: `baremetal` and `default` cause this same error when used with `-a aarch32` and `-a aarch64`.

Reply:
Yes, I get same error for `aarch32` and `aarch64`.